include/picongpu/simulation/stage/ParticleBoundaries.hpp

```diff
@@ -98,54 +98,77 @@ namespace picongpu
-                                T_Species::boundaryDescription()[d].kind = particles::boundary::Kind::Reflecting;
-                            }
-                            if(kindName == "thermal")
-                            {
-                                if(T_Species::boundaryDescription()[d].kind == particles::boundary::Kind::Periodic)
+                                        "Negative boundary offset " + errorString + " is not supported");
+                                T_Species::boundaryDescription()[d].offset = offset;
+                                float_X temperature = temperatures()[d];
+                                if(temperature < 0.0_X)
                                     throw std::runtime_error(
-                                        "Boundary kind " + errorString + " is not compatible with --thermal value");
-                                T_Species::boundaryDescription()[d].kind = particles::boundary::Kind::Thermal;
-                            }
-                            if(kindName == "periodic")
-                            {
-                                // For now it must match the default-set boundary kind
-                                if(T_Species::boundaryDescription()[d].kind != particles::boundary::Kind::Periodic)
-                                    throw std::runtime_error(
-                                        "Boundary kind " + errorString + " is not compatible with --periodic value");
-                            }
-                            if(kindName == "absorbing")
-                            {
-                                // For now it must match the default-set boundary kind
-                                if(T_Species::boundaryDescription()[d].kind != particles::boundary::Kind::Absorbing)
+                                        "Negative boundary temperature " + errorString + " is not supported");
+                                T_Species::boundaryDescription()[d].temperature = temperature;
+
+                                auto const kindName = kindNames()[d];
+                                if(kindName == "reflecting")
+                                {
+                                    if(T_Species::boundaryDescription()[d].kind == particles::boundary::Kind::Periodic)
+                                        throw std::runtime_error(
+                                            "Boundary kind " + errorString
+                                            + " is not compatible with --periodic value");
+                                    T_Species::boundaryDescription()[d].kind = particles::boundary::Kind::Reflecting;
+                                }
+                                if(kindName == "thermal")
+                                {
+                                    if(T_Species::boundaryDescription()[d].kind == particles::boundary::Kind::Periodic)
```

Comment:
There is something wrong, either here in the condition, her in the error message, or in the check above.

Reply:
What is wrong, if so we need to handle it in a separate PR, there are no changes in these lines only the indention is changed because of the additional if condition.